common/src/main/java/com/wynntils/screens/itemsharing/ItemSharingScreen.java

```diff
@@ -162,17 +184,67 @@ private void refreshPreview() {
-            Handlers.Command.sendCommand("g " + encodedItem.toUtf16String());
-        } else if (target.equals("party")) {
-            Handlers.Command.sendCommand("p " + encodedItem.toUtf16String());
-        } else if (target.equals("clipboard")) {
-            McUtils.mc().keyboardHandler.setClipboard(encodedItem.toUtf16String());
-
-            McUtils.sendMessageToClient(Component.translatable("screens.wynntils.itemSharing.copied")
-                    .withStyle(ChatFormatting.GREEN));
-        } else {
-            // TODO: Save item to storage
+        switch (target) {
+            case "guild" -> Handlers.Command.sendCommand("g " + encodedItem.toUtf16String());
+            case "party" -> Handlers.Command.sendCommand("p " + encodedItem.toUtf16String());
+            case "save" -> {
+                Map<String, Map<String, SavedItemStack>> savedItems = Models.ItemEncoding.savedItems.get();
+                Map<String, SavedItemStack> uncategorisedItems =
+                        savedItems.getOrDefault("Uncategorized", new HashMap<>());
+
+                String encodedBase64 = encodedItem.toBase64String();
+
+                // Check if the item is already saved, loop through categories then check keys (encoded item)
+                for (Map.Entry<String, Map<String, SavedItemStack>> entry : savedItems.entrySet()) {
+                    if (entry.getValue().containsKey(encodedBase64)) {
+                        McUtils.sendMessageToClient(Component.translatable(
+                                        "screens.wynntils.itemSharing.alreadySaved", itemStack.getHoverName())
+                                .withStyle(ChatFormatting.RED));
+                        return;
+                    }
+                }
+
+                ItemStack itemStackToSave = itemStack;
+
+                // Gear items can have their item changed by cosmetics so we need to get their original item
+                // FIXME: Does not work for crafted gear
+                if (wynnItem instanceof GearItem gearItem) {
+                    itemStackToSave = new FakeItemStack(gearItem, "From " + McUtils.playerName() + "'s vault");
+                }
+
+                // Leather armor can be dyed, we need to store the color
+                int color = itemStackToSave.getTag().getCompound("display").contains("color")
+                        ? itemStackToSave.getTag().getCompound("display").getInt("color")
+                        : -1;
+
+                // Regular ItemStack can't be converted to json so store the tags needed
+                // to recreate it
+                SavedItemStack savedItemStack = new SavedItemStack(
+                        Item.getId(itemStackToSave.getItem()),
+                        itemStackToSave.getTag().getInt("Damage"),
+                        itemStackToSave.getTag().getInt("HideFlags"),
+                        itemStackToSave.getTag().getBoolean("Unbreakable"),
+                        color);
```

Comment:
If you want to save non-chat encodable items, you would also need to save their lore. Not sure if this is an useful case.